exec/java-exec/src/main/resources/rest/static/js/querySubmission.js

```diff
@@ -53,7 +53,18 @@ function doSubmitQueryWithUserName() {
 //Wrap & Submit Query (invoked directly if impersonation is not enabled)
 function wrapAndSubmitQuery() {
     //Wrap if required
-    wrapQuery();
+    var mustWrapWithLimit = $('input[name="forceLimit"]:checked').length > 0;
+    //Clear field when submitting if not mustWrapWithLimit
+    if (!mustWrapWithLimit) {
+      //Wipe out any numeric entry in the field before
+      $('#autoLimit').attr('value', '');
+    } else {
+      let autoLimitValue=document.getElementById("autoLimit").value;
+      if (isNaN(autoLimitValue)) {
+        alert(autoLimitValue+ " is not a number. Please fill in a valid number");
```

Comment:
We're already using the alert mechanism for empty username (when impersonation is enabled). In both cases, we want to indicate that the user cannot progress because of a missing username or invalid number. An error message on the query page might get missed.

Reply:
ok, let's left it as is.

Reply:
This has been done as part of DRILL-7036 (#1644 )